server/bound_queries.cc

```diff
@@ -1044,7 +1291,9 @@ BoundGroupByQuery(const SelectExpression & select,
     boundRowName = rowName.bind(*groupContext);
 
     size_t maxNumRow = from.getMatrixView()->getRowCount();
-    numBuckets = maxNumRow <= 32 ? 1 : (maxNumRow / 32);
+    int numThreads = ML::num_threads();
```

Comment:
We should have around 8 * num threads buckets, to avoid a single slow thread (eg, due to it being on a core that's running a pinned or kernel task) slowing down throughput.